src/Core/Services/Implementations/UserService.cs

```diff
@@ -221,6 +221,15 @@ public override async Task<IdentityResult> DeleteAsync(User user)
                 }
             }
 
+            var onlyOwnerProviderCount = await _providerUserRepository.GetCountByOnlyOwnerAsync(user.Id);
+            if (onlyOwnerProviderCount > 0)
+            {
+                return IdentityResult.Failed(new IdentityError
+                {
+                    Description = "You must leave or delete any providers that you are the only owner of first."
```

Comment:
The `first` at the end of this sentence is bothering me. Personally, I like
```
"Cannot delete this user because it is the sole owner of at least one provider. Please delete these providers or upgrade another user."
```